lib/galaxy/managers/markdown_util.py

```diff
@@ -383,7 +384,14 @@ def _display_dataset_content(self, hda, header="Contents"):
     def handle_dataset_as_image(self, line, hda):
         dataset = hda.dataset
         name = hda.name or ''
-        with open(dataset.file_name, "rb") as f:
+        filepath = re.search(PATH_LABEL_PATTERN, line).group(2)
```

Comment:
Wouldn't this break if the path is not present (a more typical use case)? Would something like:

```
path_match = re.search(PATH_LABEL_PATTERN, line)
if path_match:
   filepath = path_match.group(2)
   file = os.path.join(hda.extra_files_path, filepath)
else:
  file = dataset.file_name
```

Make more sense?

Reply:
thanks, did it :)